### ISSUE TITLE
Add a toggle or hide directive

### ISSUE THREAD

**Issue by choldgraf** [ExecutableBookProject/sphinx-togglebutton#4]
Currently, this repository works by using classes appended to admonition / note / etc blocks. E.g.:

````
```{note}
:class: hide
My content
```
````

However, another approach could be to write our own directive that was a lightweight wrapper around a container block, so users can do something like:

````
```{hide}
**This is my content**
My content
```
````

All this would do is take the content, and call:

````
```{container} toggle
**This is my content**
My content
```
````

as a short-hand for users to remember. Maybe that would be easier to use?

**Comment by akhmerov** [comment]
How would the containers with toggle work? What would be visible in a collapsed state?

**Comment by akhmerov** [comment]
Ah, I see. Right now nothing is visible in the collapsed state. I think at least with admonitions, their header should still show up when they are collapsed.

**Comment by choldgraf** [comment]
So that's something we could control with a directive perhaps? Let people give admonition-style syntax, and decide what element to attach the toggle able piece to based on what they provide

**Comment by akhmerov** [comment]
I don't quite follow, can you make an example? Or do you mean something like

```{note-toggle}
:caption: ...
contents
```

Seems OK, but I'm unsure whether it makes the syntax easier for the users—the changes aren't too big.

**Comment by choldgraf** [comment]
What I mean is that right now, you make a content block "toggle-able" by adding a "toggle" class to it. For things like `note` directives, it looks like this:

```
.. note::
  :class: toggle
  
  some note
```

However, maybe you want to toggle something that isn't a note, or maybe you want to show the note title, but toggle the content, as you suggest. For this, there could be a specific `hide` or `toggle` directive. Something like:

```
.. toggle: toggle header
  
  Toggle body
```

If `toggle header` were provided, then it would map to an `admonition` block and the toggle would work only on the admonition *body*, not the title. If `toggle-header` were not given, then the whole content block would become hidden

**Comment by akhmerov** [comment]
Ah! This sounds really good!

I imagine the other directive parameters would be `style` (of the admonition) and a boolean `collapsed`.

**Comment by choldgraf** [comment]
This is fixed in #7
